src/sentry/snuba/tasks.py

```diff
@@ -54,7 +55,10 @@ def apply_dataset_query_conditions(dataset, query, event_types, discover=False):
     else:
         return query
 
-    return f"({event_type_conditions}) AND ({query})"
+    if query:
```

Comment:
What is this fixing?

Reply:
When `query` is empty it becomes a free text quote `()`